app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -206,17 +211,431 @@ class StateFragmentTest {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      selectMultipleChoiceOption(optionPosition = 2)
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+      clickPreviousNavigationButton()
+
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_moveFwdTwice_moveBackwdTwice_configChange_stayOnSameState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      selectMultipleChoiceOption(optionPosition = 2)
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+      clickPreviousNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
   //  7. Verifying that old answers were present when navigation backward/forward.
+  @Test
+  fun testStateFragment_loadExp_secondState_submitAnswer_moveForward_moveBackward_answerVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+
+      scrollToViewType(SUBMITTED_ANSWER)
+      onView(withId(R.id.submitted_answer_text_view)).check(
+        matches(
+          withText("1/2")
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_landscape_submitAnswer_moveForward_moveBackward_answerVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      rotateToLandscape()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      scrollToViewType(PREVIOUS_NAVIGATION_BUTTON)
+      clickPreviousNavigationButton()
+
+      scrollToViewType(SUBMITTED_ANSWER)
+      onView(withId(R.id.submitted_answer_text_view)).check(
+        matches(
+          withText("1/2")
+        )
+      )
+    }
+  }
+
   //  8. Testing providing the wrong answer and showing hints.
+  @Test
+  fun testStateFragment_loadExp_fractionState_submitWrongAnswerTwice_hintsVisible() {
```

Comment:
let's keep this under `11. Add tests for hints & solutions.` as a part of hints.

Reply:
Done